File: src/models/Order.ts
import { StringLiteralLike } from "typescript"
import { Address } from "./Address";
import { OrderItems } from "./OrderItems";
import { OrderStatus } from "./OrderStatus";

export interface Order {
    clientID: string;
    orderNumber : string;
    address: Address;
    items: OrderItems[];
    message : string
    status : OrderStatus
    date : number
}

File: app/src/main/java/com/ciejaycoding/letsmeat/models/Order.kt
package com.ciejaycoding.letsmeat.models

import android.os.Parcel
import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Order(
    var id : String? = null,
    val clientID : String? = null,
    val orderType : OrderType ? =null,
    val orderNumber : String ? =null,
    val address: Address ? = null,
    val items : List<OrderItems> ? = null,
    val message : String?= null,
    var payment: Payment ? = null,
    val date : Long ?= null,
) : Parcelable


enum class OrderType {
    DELIVER  , PICK_UP ,WALK_IN
}